notebooks/doc-preprocessing-to-sdg/utils/create_seed_dataset.py

```diff
@@ -0,0 +1,182 @@
+# Standard
+from pathlib import Path
+import json
+import re
+from typing import List, Dict
+
+# Third Party
+from datasets import Dataset, concatenate_datasets
+from transformers import AutoTokenizer
+import yaml
+
+def get_seed_dataset(path: str) -> Dataset:
+    """
+    Creates a seed dataset from a path
+    Args:
+        path (str):   Path to directory of qna.yaml and chunks
+    Returns:
+        ds (Dataset): Transformers Dataset to be used to create a jsonl
+                      of seed data for the knowledge generation pipeline in
+                      SDG.
+    """
+    valid_path = is_dir_valid(path)
+    ds = create_dataset_from_dir(valid_path)
+
+    return ds
+
+def is_dir_valid(path: str) -> Path:
+    """
+    Returns whether or not a directory contains a qna.yaml and one or more .txt chunks
+    Args:
+        path (str):       Path to directory of qna.yaml and chunks
+    Returns:
+        base_path (Path): pathlib.Path to a directory that can create a jsonl
+                          of seed data
+    """
+    base_path = Path(path)
+    if not base_path.is_dir():
+        raise ValueError("Base path must be a directory")
+
+    files = list(base_path.iterdir())
+    has_qna = any(f.name == 'qna.yaml' for f in files)
+    has_txt = any(f.suffix == '.txt' for f in files)
+    if not has_qna or not has_txt:
+        raise ValueError("Directory does not contain a qna.yaml and chunks")
+
+    return base_path
+
+def read_chunks(path: Path) -> Dict[str, str]:
+    """
+    Returns a dictionary with all of the .txt chunks in a directory
+    The chunks may originate from one or more different files
+    Args:
+        path (Path): Path to directory of chunks
+    Returns:
+        chunks_dict (Dict[str,str]: Dictionary with key of the original file name
+                                    and a list of chunks as the value
+    """
+    chunk_files = path.glob('*.txt')
+
+    chunks_dict = {}
+    for file in chunk_files:
```

Comment:
Is the file order important here? I don't think it is, but if it is, afaik `glob` lists in arbitrary order.